core/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java

```diff
@@ -384,6 +384,18 @@ private void declareParseTemplate(ExtensionDeclarer extensionDeclarer, ClassType
         .ofType(typeLoader.load(String.class))
         .withExpressionSupport(NOT_SUPPORTED)
         .describedAs("The location of the template. The order in which the processor will attempt to load the file is: from the file system, from a URL, then from the classpath.");
+
+    parseTemplate.onDefaultParameterGroup()
+        .withOptionalParameter("outputEncoding")
+        .ofType(typeLoader.load(String.class))
+        .withExpressionSupport(NOT_SUPPORTED)
```

Comment:
These parameters should support expressions

Reply:
If that is the case then not only this definition should be changed but the logic that parses the config value too. 
Check that the set-payload and set-variable processors have the same attributes and neither of those accept expressions.

Reply:
It does not make sense to add expression support for loading the template where the 'content' or 'location' that define the template are static. 
Besides that would cause the  need to reload the template from a file every time a new encoding is resolved from an expression